apps/ui/src/components/EditorVote.vue

```diff
@@ -0,0 +1,118 @@
+<script setup lang="ts">
+import Draggable from 'vuedraggable';
+import { BASIC_CHOICES, SUPPORTED_VOTING_TYPES } from '@/helpers/constants';
+import { Draft, VoteType } from '@/types';
+
+const proposal = defineModel<Draft>({ required: true });
+
+defineProps<{
+  error?: Record<number, string>;
+}>();
+
+const VOTE_TYPES = {
+  basic: {
+    label: 'Basic voting',
+    description: 'Single choice voting with three choices: For, Against or Abstain'
+  },
+  'single-choice': {
+    label: 'Single choice voting',
+    description: 'Each voter may select only one choice.'
+  },
+  approval: {
+    label: 'Approval voting',
+    description: 'Each voter may select any number of choices.'
+  }
+};
+
+function handleVoteTypeSelected(type: VoteType) {
+  if (!proposal.value) return;
+
+  proposal.value.type = type;
+
+  if (proposal.value.type === 'basic') {
+    proposal.value.choices = [...BASIC_CHOICES];
+  }
+}
+</script>
+
+<template>
+  <div class="s-base mb-5">
+    <h4 class="eyebrow mb-2.5">Voting type</h4>
+    <div class="flex flex-col gap-[12px]">
+      <button
+        v-for="(type, index) in SUPPORTED_VOTING_TYPES"
+        :key="index"
+        class="border rounded-lg p-2.5 flex gap-3 text-left"
+        :class="{ 'border-skin-text bg-skin-input-bg': proposal.type === type }"
+        @click="handleVoteTypeSelected(type as VoteType)"
```

Comment:
```suggestion
        @click="handleVoteTypeSelected(type)"
```

Reply:
Will it work if we make `SUPPORTED_VOTING_TYPES` as const?

Reply:
It works if in constants we replace 

```
export const SUPPORTED_VOTING_TYPES = ['basic', 'single-choice', 'approval'];
```

by 

```
export const SUPPORTED_VOTING_TYPES: VoteType[] = ['basic', 'single-choice', 'approval'];
```

(means we have to `import { VoteType } from '@/types';`)